File: Determine if Two Strings Are Close/determine-if-two-strings-are-close.cpp
class Solution {
public:
    bool closeStrings(string word1, string word2) {

        if (word1.length() != word2.length())
            return false;
        vector<int> w1(26, 0);
        vector<int> w2(26, 0);

        for (int i = 0; i < word1.length(); i++) {
            w1[word1[i] - 'a']++;
            w2[word2[i] - 'a']++;
        }
        for (int i = 0; i < 26; i++) {
            if (w1[i] > 0 && w2[i] == 0){
                return false;
            }
        }
        sort(w1.begin(),w1.end());
        sort(w2.begin(),w2.end());
        return w1==w2;
    }
};

File: Determine if Two Strings Are Close/determine-if-two-strings-are-close.py
class Solution(object):
    def closeStrings(self, word1, word2):
        """
        :type word1: str
        :type word2: str
        :rtype: bool
        """
        if len(word1) != len(word2):
            return False

        w1 = [0] * 26
        w2 = [0] * 26

        for c in word1:
            w1[ord(c) - ord('a')] += 1
        for c in word2:
            w2[ord(c) - ord('a')] += 1

        for i in range(26):
            if w1[i] > 0 and w2[i] == 0:
                return False

        return sorted(w1) == sorted(w2)
